### ISSUE TITLE
Consumer-Driven Contracts integration/support

### ISSUE THREAD

**Issue by x80486** [quarkusio/quarkus#6276]
**Description**

Having something like this, that works out-of-the-box without developers need to do all the Jiu Jitsu for the integration would come handy — even more for testing micro-services.

I'm thinking about integrating something like [`Pact JVM`](https://github.com/DiUS/pact-jvm/) with abilities to test **(a)** contracts for REST or HTTP as a means of communication, **(b)** message brokers (like the output from a method or a message), **(c)** you name it.

One can achieve something similar by using `JsonSchemaValidator.matchesJsonSchemaInClasspath`, but that wouldn't be a "clear way" to define the contracts:

```java
RestAssured.given()
  .when()
    .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON)
    .get(Constant.Uri.Comments)
  .then()
    .body("$.size()", Matchers.greaterThanOrEqualTo(4))
    .body(JsonSchemaValidator.matchesJsonSchemaInClasspath("schema/json/comments__get-all.json"))
    .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
    .statusCode(HttpStatus.SC_OK);
```

**Comment by bpechiney** [comment]
Has there been any work done to support contract testing in Quarkus?

**Comment by dicolasi** [comment]
Same question here :)

**Comment by ambition-consulting** [comment]
It's a shame that a framework, which is meant for microservice orchestration, has only testing support for monolithic applications. We need orchestration support like provided by spring-cloud-contracts or pact.

**Comment by wabrit** [comment]
I would agree with @ambition-consulting - CDC is a big part of our Spring Boot microservice qa process, and it would be very desirable to have an equivalent in quarkus. Ideally the framework should be language-agnostic so that e.g. contracts published by a Quarkus microservice producer can be tested by a Nodejs consumer - spring-cloud-contracts AFAIK does not support that.

**Comment by ambition-consulting** [comment]
Apparently Pact tests can at least get executed again with Quarkus 2: https://github.com/skattela/pact-workshop-jvm-quarkus

**Comment by holly-cummins** [comment]
See #27729, which is arguably a duplicate of this, but with the extension-proposal type.

**Comment by holly-cummins** [comment]
See also #9677 which sets many pact artifacts to be parentfirst in a core pom, with a comment 
```
                        <!-- Pact[<4.1.0] support. It would be great if there was somewhere better to put this-->
```
Those updates should perhaps? move to this extension once implemented.

**Comment by holly-cummins** [comment]
See https://github.com/quarkiverse/quarkus-pact/, and also https://www.youtube.com/watch?v=d9CSY8HuZ9U, which discusses some of the roadmap for CDC support in Quarkus.

**Comment by holly-cummins** [comment]
I'm going to close this as satisfied by https://quarkus.io/extensions/io.quarkiverse.pact/quarkus-pact-consumer/ and https://quarkus.io/extensions/io.quarkiverse.pact/quarkus-pact-provider/
